src/Config.ts

```diff
@@ -1,30 +1,76 @@
-import { open } from "node:fs/promises";
+import { open, stat, access, constants } from "node:fs/promises";
 import Logger from "./Logger.js";
 
-export default class Config {
-    public port: number = 25565;
+export interface Config {
+    /**
+     * Port to listen on
+     */
+    port: number;
 
     /**
      * Kick reason for when the server is shutting down
      */
-    public shutdownKickReason: string = "Server closed";
+    shutdownKickReason: string;
+}
 
+export class ConfigLoader {
     /**
      * Get a Config instance from a json file
      * @param file The file to read from
      * @returns a promise that resolves to a Config instance
      */
     public static async fromFile(file: string): Promise<Config> {
+        if (!(await ConfigLoader.exists(file))) {
+            new Logger("Config").warn("Config does not exist, creating default '%s'", file);
+            await ConfigLoader.createDefault(file);
+            return ConfigLoader.getDefault();
+        }
+
         try {
             const fd = await open(file, "r");
             const data = await fd.readFile("utf-8");
             const config = JSON.parse(data) as Config;
             fd.close();
             return config;
         } catch {
-            new Logger("Config").error("Failed to read config file, using default config");
-            return new Config();
+            new Logger("Config").error("Failed to read '%s', check your file permissions.", file);
+            process.exit(1); //TODO: better exit handling
         }
     }
 
+    /**
+     * Get a default config instance
+     * @returns a default config instance
+    **/
+    public static getDefault(): Config {
+        return {
+            port: 25565,
+            shutdownKickReason: "Server shutting down"
+        };
+    }
+
+    /**
+     * Checks if a config exists
+     * @param file The file to check
+     * @returns a promise that resolves to a boolean
+     */
+    public static async exists(file: string): Promise<boolean> {
+        try {
+            await stat(file);
+            access(file, constants.R_OK).catch(() => void 0);
```

Comment:
```suggestion
            await access(file, constants.F_OK);
```
we should use this to replace stat as it should be more efficient (`stat` checks and returns a bunch of data, `access F.OK` just checks if it exists)